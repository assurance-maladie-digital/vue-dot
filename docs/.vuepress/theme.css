html, body {
	font-size: 100% !important;
}

.v-ripple__container,
.v-ripple__container * {
	max-width: none;
}

.v-input--selection-controls__ripple {
	max-width: none;
}

.application--wrap {
	width: 100%;
}

.theme--light.v-input--is-disabled .v-label,
.theme--light.v-icon.v-icon--disabled,
.theme--light.v-list .v-list--disabled {
	color: #767676 !important;
}

.theme--light.v-btn.v-btn--disabled .v-icon {
	color: #949494 !important;
}

.v-tabs__bar .v-tabs__item--disabled {
	color: #333 !important;
}

.theme--light.v-btn.v-btn--disabled {
	color: #646464 !important;
}

.content h1 {
	font-size: 2.2rem;
	margin-bottom: 25px;
}

.content h2 {
	font-size: 1.65rem;
	margin-bottom: 16px;
	margin-top: 35px;
}

.content h2 + h3 {
	margin-top: 0;
}

.content h3 {
	font-size: 1.35rem;
	margin-bottom: 10px;
	margin-top: 35px;
}

.content h3 + h4 {
	margin-top: 0;
}

.content h4 {
	font-size: 1.15rem;
	margin-bottom: 5px;
	margin-top: 15px;
}

pre[class*='language-'] {
	align-items: center;
	box-shadow: none;
	display: flex;
	border-radius: 2px;
	position: relative;
	overflow-x: auto;
	overflow-y: hidden;
	background: #2d2d2d;
	margin: 0;
	margin-bottom: 5px;
}

code[class*='language-'], code {
	position: relative;
	box-shadow: none;
	overflow-x: auto;
	overflow-y: hidden;
	word-break: break-word;
	flex-wrap: wrap;
	align-items: center;
	vertical-align: middle;
	white-space: pre-wrap;
	margin: 0;
	font-family: 'Inconsolata', monospace;
	font-weight: 300;
	font-size: .9rem;
	line-height: 1.55;
	padding: 0 5px;
}

code {
	background: rgba(240, 240, 240, .8);
	color: rgba(0, 0, 0, .7);
}

code::after,
code::before {
	content: none;
}

pre code {
	background: transparent;
	padding: 0;
	color: #ccc;
}

.header-anchor {
	color: #0c4887;
	transition: opacity .15s !important;
	text-decoration: none;
	font-size: .85em;
	float: left;
	margin-left: -.8em;
	margin-top: .125em;
	padding-right: .23em;
	opacity: 0;
	position: absolute;
}

[id]:hover > .header-anchor,
.header-anchor:focus {
	opacity: 1;
}

.icon.outbound {
	color: inherit;
}

.custom-block {
	padding: .1rem 1.5rem;
	border-left-width: .5rem;
	border-left-style: solid;
	margin: 1rem 0;
	background-color: #f3f5f7 !important;
	border-color: #42b983;
}

.custom-block.warning {
	background-color: rgba(255, 229, 100, .3) !important;
	border-color: #e7c000 !important;
}

.custom-block .custom-block-title {
	font-weight: 600;
	margin-top: 10px;
	margin-bottom: 0;
}

.custom-block.warning .custom-block-title {
	color: #776300;
}

.custom-block.danger {
	background-color: #ffe6e6 !important;
	border-color: #c00 !important;
}

.custom-block.danger .custom-block-title {
	color: #4d0000;
}
