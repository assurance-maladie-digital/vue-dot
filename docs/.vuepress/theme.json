{
	"name": "digital",
	"version": "0.0.1",
	"config": {
		"styles": {
			"html, body": {
				"font-size": "100% !important"
			},
			".v-ripple__container, .v-ripple__container *": {
				"max-width": "none"
			},
			".v-input--selection-controls__ripple": {
				"max-width": "none"
			},
			".application--wrap": {
				"width": "100%"
			},
			".theme--light.v-input--is-disabled .v-label, .theme--light.v-icon.v-icon--disabled, .theme--light.v-list .v-list--disabled": {
				"color": "#767676 !important"
			},
			".theme--light.v-btn.v-btn--disabled .v-icon": {
				"color": "#949494 !important"
			},
			".v-tabs__bar .v-tabs__item--disabled": {
				"color": "#333 !important"
			},
			".theme--light.v-btn.v-btn--disabled": {
				"color": "#646464 !important"
			},
			".content h1": {
				"font-size": "2.2rem",
				"margin-bottom": "25px"
			},
			".content h2": {
				"font-size": "1.65rem",
				"margin-bottom": "16px",
				"margin-top": "35px"
			},
			".content h2 + h3": {
				"margin-top": "0"
			},
			".content h3": {
				"font-size": "1.35rem",
				"margin-bottom": "10px",
				"margin-top": "35px"
			},
			".content h3 + h4": {
				"margin-top": "0"
			},
			".content h4": {
				"font-size": "1.15rem",
				"margin-bottom": "5px",
				"margin-top": "15px"
			},
			"pre[class*='language-']": {
				"align-items": "center",
				"box-shadow": "none",
				"display": "flex",
				"border-radius": "2px",
				"position": "relative",
				"overflow-x": "auto",
				"overflow-y": "hidden",
				"background": "#2d2d2d",
				"margin": "0",
				"margin-bottom": "5px"
			},
			"code[class*='language-'], code": {
				"position": "relative",
				"box-shadow": "none",
				"overflow-x": "auto",
				"overflow-y": "hidden",
				"word-break": "break-word",
				"flex-wrap": "wrap",
				"align-items": "center",
				"vertical-align": "middle",
				"white-space": "pre-wrap",
				"margin": "0",
				"font-family": "'Inconsolata', monospace",
				"font-weight": "300",
				"font-size": ".9rem",
				"line-height": "1.55",
				"padding": "0 5px"
			},
			"code": {
				"background": "rgba(240, 240, 240, .8)",
				"color": "rgba(0, 0, 0, .7)"
			},
			"code::after, code::before": {
				"content": "none"
			},
			"pre code": {
				"background": "transparent",
				"padding": "0",
				"color": "#ccc"
			},
			".header-anchor": {
				"color": "#0c4887",
				"transition": "opacity .15s",
				"text-decoration": "none",
				"font-size": ".85em",
				"float": "left",
				"margin-left": "-.8em",
				"margin-top": ".125em",
				"padding-right": ".23em",
				"opacity": "0",
				"position": "absolute"
			},
			"[id]:hover > .header-anchor, .header-anchor:focus": {
				"opacity": "1"
			},
			".icon.outbound": {
				"color": "inherit"
			},
			".custom-block": {
				"padding": ".1rem 1.5rem",
				"border-left-width": ".5rem",
				"border-left-style": "solid",
				"margin": "1rem 0",
				"background-color": "#f3f5f7 !important",
				"border-color": "#42b983"
			},
			".custom-block.warning": {
				"background-color": "rgba(255, 229, 100, .3) !important",
				"border-color": "#e7c000 !important"
			},
			".custom-block .custom-block-title": {
				"font-weight": "600",
				"margin-top": "10px",
				"margin-bottom": "0"
			},
			".custom-block.warning .custom-block-title": {
				"color": "#776300"
			},
			".custom-block.danger": {
				"background-color": "#ffe6e6 !important",
				"border-color": "#c00 !important"
			},
			".custom-block.danger .custom-block-title": {
				"color": "#4d0000"
			}
		},
		"colors": {
			"primary": "#0c4887",
			"secondary": "#01579b",
			"accent": "#0079bc",
			"error": "#e54d42",
			"info": "#0077c5",
			"success": "#2d9437",
			"warning": "#977000"
		},
		"icons": [
			{
				"name": "vuejs",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 400 400\"><path fill=\"#4dba87\" d=\"M237.42 86.66L207.19 139l-30.22-52.35H76.3l130.9 226.69L338.07 86.66z\"/><path fill=\"#435466\" d=\"M237.42 86.66L207.19 139l-30.22-52.35h-48.3l78.52 136 78.53-136z\"/></svg>"
			},
			{
				"name": "vue-dot-dark",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 400 400\"><path fill=\"#fff\" d=\"M184.55 78.53H61.8l145.4 251.84L352.6 78.53H229.84l-22.64 39.21-22.64-39.21z\"/><path fill=\"#1867c0\" d=\"M237.42 86.66L207.19 139l-30.22-52.35H76.3L207.2 313.34 338.07 86.66z\"/><path fill=\"#1697f6\" d=\"M237.42 86.66L207.19 139l-30.22-52.35h-48.3l78.52 136 78.53-136z\"/></svg>"
			},
			{
				"name": "github",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 24 24\" fill=\"currentColor\"><path d=\"M22.39 6.27A11.95 11.95 0 0 0 12 .29a11.95 11.95 0 0 0-12 12c0 2.62.76 4.97 2.29 7.06a11.75 11.75 0 0 0 5.91 4.33c.28.06.5.02.63-.1.13-.13.2-.29.2-.47a359.71 359.71 0 0 0-.01-2.24l-.36.06a6.6 6.6 0 0 1-1.95-.05 2.43 2.43 0 0 1-1.05-.47 1.98 1.98 0 0 1-.7-.96l-.15-.36a3.9 3.9 0 0 0-.49-.8c-.22-.29-.45-.49-.68-.59l-.1-.08a1.15 1.15 0 0 1-.35-.4c-.03-.08 0-.14.08-.18.08-.05.23-.07.45-.07l.31.04c.2.04.47.17.77.38.31.2.56.48.76.8.24.44.53.76.87.99.34.22.68.33 1.02.33.35 0 .64-.02.9-.08.24-.05.48-.13.7-.23.09-.7.35-1.24.76-1.61a10.7 10.7 0 0 1-1.6-.28 6.38 6.38 0 0 1-1.47-.61A4.2 4.2 0 0 1 4.66 14a7.8 7.8 0 0 1-.32-2.34c0-1.26.42-2.34 1.24-3.22a4.2 4.2 0 0 1 .1-3.19c.3-.1.76-.02 1.35.21A9.4 9.4 0 0 1 9 6.5a11.1 11.1 0 0 1 6 0l.6-.38c.4-.25.88-.48 1.43-.69.55-.2.98-.26 1.27-.17.47 1.18.5 2.24.12 3.19a4.56 4.56 0 0 1 1.24 3.22c0 .88-.11 1.67-.32 2.35a4.84 4.84 0 0 1-.83 1.64c-.34.41-.76.76-1.27 1.04-.5.28-1 .48-1.47.6-.47.13-1 .23-1.6.29.54.47.81 1.2.81 2.22v3.3c0 .18.07.34.2.46.13.13.34.17.62.11 2.41-.8 4.39-2.24 5.91-4.33A11.66 11.66 0 0 0 24 12.29c0-2.17-.54-4.18-1.61-6.02z\"/></svg>"
			},
			{
				"name": "dashboard",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 24 24\" fill=\"currentColor\"><path d=\"M13.33 0H24v8H13.33zM0 0h10.67v13.33H0zM0 16h10.67v8H0zM13.33 10.67H24V24H13.33z\"/></svg>"
			},
			{
				"name": "package",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 24 24\" fill=\"currentColor\"><path d=\"M13.97 12.19l-1.07.62V24l9.68-5.6V7.23zM2.05 5.75L5.78 7.9l9.95-5.74L12 0zM12 11.5l3.7-2.14 6.25-3.61-4.3-2.45-9.94 5.75.57.3zM5.17 9.4L1.5 7.28v11.1l9.61 5.56v-11.1l-4.15-2.4v2.72l-1.79-.92z\"/></svg>"
			},
			{
				"name": "function",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 24 24\" fill=\"currentColor\"><path d=\"M15.75 0S12.6 0 11.7 4.5l-.6 3H8.25L7.5 9h3.3L8.7 19.5c-.6 3-2.85 3-2.85 3h-1.5l-.6 1.5h4.5s3.15 0 4.05-4.5L14.4 9H18l.75-1.5H14.7l.6-3c.6-3 2.7-3 2.7-3h1.5l.75-1.5z\"/></svg>"
			},
			{
				"name": "codepen",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 24 24\" fill=\"currentColor\"><path d=\"M12.9.32a1.9 1.9 0 0 0-2.14.02L1.23 6.98C.72 7.34.4 7.92.4 8.54v7.16c0 .63.32 1.23.85 1.58l9.7 6.4a1.9 1.9 0 0 0 2.09 0l9.88-6.53c.54-.35.86-.94.86-1.58V8.54a1.9 1.9 0 0 0-.85-1.58zm7.38 7.75l-3.66 2.45-3.42-2.49-.03-4.9zM10.7 3.13v4.8l-3.52 2.46-3.5-2.35zM2.72 10.4L5 11.92l-2.27 1.59zm8.04 10.23l-7.08-4.75 3.54-2.46 3.54 2.37zM9.4 11.88l2.3-1.6 2.66 1.75L12 13.62zm3.84 8.75v-4.84l3.39-2.27 3.64 2.4zm8.04-7.04L18.89 12l2.4-1.61z\"/></svg>"
			},
			{
				"name": "vuetify",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 24 24\"><path fill=\"#1697f6\" d=\"M12 0H7.08L12 11.64\"/><path fill=\"#1867c0\" d=\"M12 0h4.93L12 11.64\"/><path fill=\"#7bc6ff\" d=\"M12 15v9L1.5 3.5H7\"/><path fill=\"#aeddff\" d=\"M12 15v9L22.5 3.5H17\"/></svg>"
			},
			{
				"name": "earth",
				"svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 24 24\" fill=\"currentColor\"><path d=\"M20.49 3.51A11.92 11.92 0 0 1 24 12c0 3.2-1.25 6.22-3.51 8.49A11.92 11.92 0 0 1 12 24c-3.2 0-6.22-1.25-8.49-3.51A11.92 11.92 0 0 1 0 12c0-3.2 1.25-6.22 3.51-8.49A11.92 11.92 0 0 1 12 0c3.2 0 6.22 1.25 8.49 3.51zm-1.07 15.91a10.43 10.43 0 0 0 2.87-5.32c-.32.47-.62.64-.8-.4-.2-1.7-1.76-.61-2.74-1.22-1.03.7-3.34-1.35-2.95.96.6 1.04 3.28-1.4 1.95.81-.85 1.54-3.11 4.94-2.82 6.71.04 2.57-2.63.54-3.54-.32-.62-1.7-.21-4.7-1.83-5.53-1.75-.08-3.26-.24-3.94-2.2-.4-1.4.44-3.49 1.94-3.8 2.2-1.39 2.99 1.61 5.05 1.67.64-.67 2.39-.89 2.53-1.64C13.8 8.9 16.86 8 15.01 7.5c-1.02.12-1.67 1.06-1.13 1.85-1.98.46-2.04-2.85-3.94-1.8-.05 1.65-3.1.53-1.05.2.7-.31-1.15-1.2-.15-1.04.49-.03 2.14-.6 1.7-1 .91-.57 1.68 1.37 2.58-.04.65-1.08-.27-1.28-1.08-.73-.46-.52.8-1.63 1.93-2.1.37-.17.73-.25 1-.23.56.65 1.6.76 1.66-.08a10.46 10.46 0 0 0-10.8 1.05c.49.22.76.5.3.85C5.67 5.5 4.2 6.95 2.9 6.75a10.38 10.38 0 0 0-1.3 3.76c1.08.36 1.33 1.06 1.1 1.3a3.55 3.55 0 0 0-1.07 1.9 10.42 10.42 0 0 0 2.94 5.71A10.43 10.43 0 0 0 12 22.5c2.8 0 5.44-1.1 7.42-3.08z\"/></svg>"
			}
		]
	}
}
