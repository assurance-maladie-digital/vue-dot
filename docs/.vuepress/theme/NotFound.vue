<template>
	<VApp>
		<Toolbar />

		<VContent>
			<VContainer>
				 <VLayout align-center>
					<VFlex text-xs-center>
						<h1 class="display-2 primary--text mt-5 mb-3">{{ $t('error404.title') }}</h1>

						<p>{{ $t('error404.description') }}</p>

						<VBtn
							:to="`/${this.$i18n.locale}/guide/`"
							color="primary"
							outline
							exact
						>
							{{ $t('error404.button') }}
						</VBtn>
					</VFlex>
				</VLayout>
			</VContainer>
		</VContent>
	</VApp>
</template>

<script>
	import Toolbar from '../components/Toolbar.vue';

	export default {
		name: 'NotFound',
		components: {
			Toolbar
		},
		mounted() {
			// Redirect default route
			if (this.$route.path === '/') {
				this.$router.push(`/${this.$i18n.locale}/`);
			}
		}
	};
</script>
