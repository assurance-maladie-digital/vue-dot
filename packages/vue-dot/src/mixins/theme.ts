import Vue from 'vue';

const theme = {
	created(this: Vue) {
		if (this.$theme && this.$theme.config.colors) {
			this.$vuetify.theme = this.$theme.config.colors;
		}
	}
};

export default theme;
