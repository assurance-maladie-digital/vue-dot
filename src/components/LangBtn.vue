<template>
	<VMenu
		:absolute="absolute"
		:allow-overflow="allowOverflow"
		:auto="auto"
		:bottom="menuBottom"
		:close-delay="closeDelay"
		:close-on-click="closeOnClick"
		:close-on-content-click="closeOnContentClick"
		:content-class="contentClass"
		:dark="dark"
		:disable-keys="disableKeys"
		:disabled="disabled"
		:fixed="fixed"
		:full-width="fullWidth"
		:lazy="lazy"
		:left="menuLeft"
		:light="light"
		:max-height="maxHeight"
		:max-width="maxWidth"
		:min-width="minWidth"
		:nudge-bottom="nudgeBottom"
		:nudge-left="nudgeLeft"
		:nudge-right="nudgeRight"
		:nudge-top="nudgeTop"
		:nudge-width="nudgeWidth"
		:offset-overflow="offsetOverflow"
		:offset-x="offsetX"
		:offset-y="offsetY"
		:open-delay="openDelay"
		:open-on-click="openOnClick"
		:open-on-hover="openOnHover"
		:origin="origin"
		:position-x="positionX"
		:position-y="positionY"
		:right="menuRight"
		:top="menuTop"
		:transition="transition"
		:value="menuValue"
		:z-index="zIndex"
	>
		<VBtn
			slot="activator"
			:aria-label="`${label} ${languages[currentLangUsed].nativeName}`"

			:absolute="absolute"
			:active-class="activeClass"
			:append="append"
			:block="block"
			:bottom="bottom"
			:color="color"
			:dark="dark"
			:depressed="depressed"
			:disabled="disabled"
			:fab="fab"
			:fixed="fixed"
			:flat="flat"
			:icon="icon"
			:large="large"
			:left="left"
			:light="light"
			:loading="loading"
			:outline="outline"
			:right="right"
			:ripple="ripple"
			:round="round"
			:small="small"
			:tag="tag"
			:top="top"
			:type="type"
			:value="btnValue"
		>
			<img
				v-if="flags"
				:src="`${flagsUrl}${currentLangUsed}.svg`"
				class="current-flag"
				alt=""
			>

			<span
				v-if="displayTextBtn"
				class="ml-2"
			>
				{{ languages[currentLangUsed].nativeName }}
			</span>

			<SvgIcon
				v-if="displayArrow"
				size="10px"
				class="ml-2"
				icon="expand"
			/>
		</VBtn>

		<VList
			v-if="languages"

			:dark="dark"
			:dense="dense"
			:light="light"
		>
			<VListTile
				v-for="(item, lang, index) in languages"
				:key="index"

				:active-class="listActiveClass"
				:avatar="avatar"
				:color="color"
				:dark="dark"
				:disabled="disabled"
				:light="light"
				:ripple="ripple"
				:tag="listTag"
			>
				<VBtn
					:color="color"
					flat
					class="ma-0 text-none trigger-btn"
					@click="changeLang(lang)"
				>
					<VListTileAvatar v-if="flags">
						<img
							:src="`${flagsUrl}${lang}.svg`"
							class="flag-img"
							alt=""
						>
					</VListTileAvatar>

					<VListTileTitle>{{ item.nativeName }}</VListTileTitle>
				</VBtn>
			</VListTile>
		</VList>
	</VMenu>
</template>

<script lang="ts">
	import Vue from 'vue';

	import languages from 'languages';

	export default Vue.extend({
		name: 'LangBtn',
		model: {
			prop: 'value',
			event: 'change'
		},
		props: {
			// Global
			availableLanguages: {
				type: [Object, Array, String],
				default: '*'
			},
			label: {
				type: String,
				default: 'Choix de la langue. Actuellement :'
			},
			flags: {
				type: Boolean,
				default: false
			},
			flagsUrl: {
				type: String,
				default: '/img/flags/'
			},
			displayTextBtn: {
				type: Boolean,
				default: true
			},
			displayArrow: {
				type: Boolean,
				default: true
			},
			value: {
				type: String,
				default: 'fr'
			},
			// Shared
			absolute: {
				type: Boolean,
				default: false
			},
			dark: {
				type: Boolean,
				default: false
			},
			disabled: {
				type: Boolean,
				default: false
			},
			fixed: {
				type: Boolean,
				default: false
			},
			light: {
				type: Boolean,
				default: false
			},
			// Menu
			allowOverflow: {
				type: Boolean,
				default: true
			},
			auto: {
				type: Boolean,
				default: false
			},
			closeDelay: {
				type: [Number, String],
				default: 0
			},
			closeOnClick: {
				type: Boolean,
				default: true
			},
			closeOnContentClick: {
				type: Boolean,
				default: true
			},
			contentClass: {
				type: String,
				default: undefined
			},
			disableKeys: {
				type: Boolean,
				default: false
			},
			fullWidth: {
				type: Boolean,
				default: false
			},
			lazy: {
				type: Boolean,
				default: false
			},
			maxHeight: {
				type: [Number, String],
				default: 'auto'
			},
			maxWidth: {
				type: [Number, String],
				default: 'auto'
			},
			minWidth: {
				type: [Number, String],
				default: undefined
			},
			nudgeBottom: {
				type: [Number, String],
				default: 0
			},
			nudgeLeft: {
				type: [Number, String],
				default: 0
			},
			nudgeRight: {
				type: [Number, String],
				default: 0
			},
			nudgeTop: {
				type: [Number, String],
				default: 0
			},
			nudgeWidth: {
				type: [Number, String],
				default: 0
			},
			offsetOverflow: {
				type: Boolean,
				default: false
			},
			offsetX: {
				type: Boolean,
				default: false
			},
			offsetY: {
				type: Boolean,
				default: true
			},
			openDelay: {
				type: [Number, String],
				default: 0
			},
			openOnClick: {
				type: Boolean,
				default: true
			},
			openOnHover: {
				type: Boolean,
				default: false
			},
			origin: {
				type: String,
				default: 'top left'
			},
			positionX: {
				type: Number,
				default: undefined
			},
			positionY: {
				type: Number,
				default: undefined
			},
			transition: {
				type: [Boolean, String],
				default: 'v-menu-transition'
			},
			menuValue: {
				type: [Boolean, Object, Number, String],
				default: undefined
			},
			zIndex: {
				type: [Number, String],
				default: undefined
			},
			menuLeft: {
				type: Boolean,
				default: false
			},
			menuBottom: {
				type: Boolean,
				default: false
			},
			menuRight: {
				type: Boolean,
				default: false
			},
			menuTop: {
				type: Boolean,
				default: false
			},
			// Btn
			activeClass: {
				type: String,
				default: 'v-btn--active'
			},
			append: {
				type: Boolean,
				default: false
			},
			block: {
				type: Boolean,
				default: false
			},
			color: {
				type: String,
				default: 'currentColor'
			},
			depressed: {
				type: Boolean,
				default: true
			},
			fab: {
				type: Boolean,
				default: false
			},
			flat: {
				type: Boolean,
				default: false
			},
			icon: {
				type: Boolean,
				default: false
			},
			large: {
				type: Boolean,
				default: false
			},
			loading: {
				type: Boolean,
				default: false
			},
			outline: {
				type: Boolean,
				default: true
			},
			ripple: {
				type: [Boolean, Object],
				default: true
			},
			round: {
				type: Boolean,
				default: false
			},
			small: {
				type: Boolean,
				default: false
			},
			tag: {
				type: String,
				default: 'button'
			},
			type: {
				type: String,
				default: 'button'
			},
			btnValue: {
				type: [String, Number, Boolean],
				default: undefined
			},
			left: {
				type: Boolean,
				default: false
			},
			bottom: {
				type: Boolean,
				default: false
			},
			right: {
				type: Boolean,
				default: false
			},
			top: {
				type: Boolean,
				default: false
			},
			// List
			avatar: {
				type: Boolean,
				default: true
			},
			dense: {
				type: Boolean,
				default: false
			},
			listActiveClass: {
				type: String,
				default: 'primary--text'
			},
			listTag: {
				type: String,
				default: undefined
			}
		},
		data(this: any): any {
			return {
				languagesData: this.getLanguagesWithData(),
				currentLangUsed: this.value || 'fr'
			};
		},
		computed: {
			languages(): object {
				let data: any = {};

				if (this.availableLanguages !== '*') {
					this.availableLanguages.forEach((lang: string) => {
						data[lang] = this.languagesData[lang];
					});
				} else {
					data = this.languagesData;
				}

				return data;
			}
		},
		mounted() {
			this.$emit('change', this.currentLangUsed);
		},
		methods: {
			changeLang(lang: string): void {
				this.$emit('change', lang);
				this.currentLangUsed = lang;
			},
			getLanguagesWithData(): object[] {
				const data: any = {};

				languages.getAllLanguageCode().forEach((lang: string) => {
					const obj = languages.getLanguageInfo(lang);
					data[lang] = obj;
				});

				return data;
			}
		}
	});
</script>

<style lang="scss" scoped>
	.v-avatar img {
		border-radius: 0%;
	}

	.v-menu__content {
		overflow: auto;
		max-height: 300px;
	}

	.v-list {
		.v-btn {
			width: 100%;
			height: 100%;
		}

		/deep/ .v-list__tile {
			padding: 0;
		}
	}

	.current-flag {
		width: 30px;
	}
</style>
