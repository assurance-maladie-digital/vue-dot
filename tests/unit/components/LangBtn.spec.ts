import Vue from 'vue';
import { mount, createLocalVue } from '@vue/test-utils';
import LangBtn from 'vue-dot-src/components/LangBtn.vue';

import Vuetify from 'vuetify';
import VueDot from 'vue-dot-src/index';
import { default as theme } from '../theme.json';

const localVue = createLocalVue();
Vue.use(Vuetify);

localVue.use(VueDot, {
	theme
});

describe('LangBtn.vue', () => {
	const build = () => {
		document.body.setAttribute('data-app', 'true');

		const wrapper = mount(LangBtn, {
			localVue,
			propsData: {
				value: ''
			}
		});

		return wrapper;
	};

	it('renders correctly', () => {
		const wrapper = build();

		expect((wrapper.vm as any).currentLangUsed).toBe('fr');

		expect(wrapper.html()).toMatchSnapshot();
	});

	it('renders available languages', () => {
		const wrapper = build();
		wrapper.setProps({
			availableLanguages: [
				'fr',
				'en'
			]
		});

		expect(wrapper.html()).toMatchSnapshot();
	});

	it('updates the language when clicked', () => {
		const wrapper = build();

		const btn = wrapper.find('.trigger-btn');
		btn.trigger('click');

		expect((wrapper.vm as any).currentLangUsed).toBe('aa');
	});
});
