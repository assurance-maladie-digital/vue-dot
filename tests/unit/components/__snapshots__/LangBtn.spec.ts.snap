// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LangBtn.vue renders available languages 1`] = `
<div class="v-menu lang-btn v-menu--inline">
  <div class="v-menu__activator"><button type="button" class="v-btn v-btn--outline v-btn--depressed theme--light currentColor--text" aria-label="Choix de la langue. Actuellement : Français">
      <div class="v-btn__content">
        <!----> <span class="ml-2">
			Français
		</span> <span aria-hidden="true" class="x-svg-icon ml-2 custom-color" style="width: 10px; height: 10px;"><svg viewBox="0 0 24 24"><path d="M24 7.4l-2.8-2.8-9.2 9.2-9.2-9.2L0 7.4l12 12z"></path></svg> </span></div>
    </button></div>
</div>
`;

exports[`LangBtn.vue renders correctly 1`] = `
<div class="v-menu lang-btn v-menu--inline">
  <div class="v-menu__activator"><button type="button" class="v-btn v-btn--outline v-btn--depressed theme--light currentColor--text" aria-label="Choix de la langue. Actuellement : Français">
      <div class="v-btn__content">
        <!----> <span class="ml-2">
			Français
		</span> <span aria-hidden="true" class="x-svg-icon ml-2 custom-color" style="width: 10px; height: 10px;"><svg viewBox="0 0 24 24"><path d="M24 7.4l-2.8-2.8-9.2 9.2-9.2-9.2L0 7.4l12 12z"></path></svg> </span></div>
    </button></div>
</div>
`;
